staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -64,7 +64,12 @@ func (w *realTimeoutFactory) TimeoutCh() (<-chan time.Time, func() bool) {
 // serveWatch will serve a watch response.
 // TODO: the functionality in this method and in WatchServer.Serve is not cleanly decoupled.
 func serveWatch(watcher watch.Interface, scope *RequestScope, mediaTypeOptions negotiation.MediaTypeOptions, req *http.Request, w http.ResponseWriter, timeout time.Duration, metricsScope string) {
-	defer watcher.Stop()
+	closeWatch := true
```

Comment:
Let's not move this part here yet - if for whatever reason we decide to not proceed with the main change, this isn't desired change.

Reply:
Done